### ISSUE TITLE
Cylc hub sort ordering doesn't work for all columns

### ISSUE THREAD

**Issue by ColemanTom** [cylc/cylc-uiserver#577]
NOTE: From what I can see, the below report is using cylc-uiserver-base=1.4.3 and cylc-uiserver-hub-base=1.4.3 (latest versions) and cylc-ui=2.3.0.

### Description

I have accessed the `Cylc Hub` via the UI-server, and the default listing shows this information in the `Last activity` column:

![image](https://github.com/cylc/cylc-uiserver/assets/15375218/4561845c-8b68-4733-b599-e99d8d0528f7)

If I click on the sorting for that column, I get:

![image](https://github.com/cylc/cylc-uiserver/assets/15375218/a568e163-afbe-4bc5-a95a-c9f836c7e751)

Click again I get zero changes, 

![image](https://github.com/cylc/cylc-uiserver/assets/15375218/a568e163-afbe-4bc5-a95a-c9f836c7e751)

Click again and I get the original order

![image](https://github.com/cylc/cylc-uiserver/assets/15375218/4561845c-8b68-4733-b599-e99d8d0528f7)

I cannot figure out how it is sorting. It doesn't seem to be by the numeric number, nor as a datetime offset. For example, in the initial/final view, I can see 9 days, 11 minutes, 9 days. In the middle views, I can see Never, 5 days, Never.

### Expected Behaviour

I would personally expect a datetime offset sort order.

**Comment by hjoliver** [comment]
Looks like this problem: https://github.com/jupyterhub/jupyterhub/issues/3816

Merged to JupyterHub main branch recently, but not released yet.

**Comment by oliver-sanders** [comment]
Yep, the page you are looking at comes from Jupyter Hub, not Cylc.

Note, that "Cylc Hub" is really just a configuration of Jupyter Hub that is configured to launch the Cylc UI Server by default. We expose this behind the Cylc CLI to make it more approachable to Cylc users, but it's just Jupyter Hub really.
